crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -390,6 +390,7 @@ impl<F: Send + Clone, Ctx: PaymentMethodRetrieve>
             setup_mandate,
             customer_acceptance,
             token,
+            token_data: None,
```

Comment:
In case of confirm `true` shouldn't we fetch the token_data form redis and pass it in the session object?

Reply:
No. In case of create, we do not support token based payments.